iceoryx_meta/build_options.cmake

```diff
@@ -29,6 +29,7 @@ option(DOWNLOAD_TOML_LIB "Download cpptoml via the CMake ExternalProject module"
 option(EXAMPLES "Build all iceoryx examples" OFF)
 option(INTROSPECTION "Builds the introspection client which requires the ncurses library with an activated terminfo feature" OFF)
 option(ONE_TO_MANY_ONLY "Restricts communication to 1:n pattern" OFF)
+option(PLATFORM_PATH "Overrides integrated platform detection and uses provided custom path" OFF)
```

Comment:
I think it is not a good idea to mix a CMake option (which is treated as boolean) with a CMake argument (given as string value).

For me the `PLATFORM_PATH` look similar to the `CMAKE_TOOLCHAIN_FILE` argument which we also treat not as option.

What you need to do is to check if the variable is set and have a value (two different things in CMake)

Maybe something like this could work (not tested):
```
if(DEFINED PREFIX OR NOT PREFIX STREQUAL "")
```

Reply:
I used `set(PLATFORM_PATH "" CACHE PATH "description")` which solves also the relative path issue.